sentry-android-core/src/main/java/io/sentry/android/core/EnvelopeFileObserver.java

```diff
@@ -0,0 +1,39 @@
+package io.sentry.android.core;
+
+import android.os.FileObserver;
+import io.sentry.core.*;
+import io.sentry.core.util.Objects;
+
+final class EnvelopeFileObserver extends FileObserver {
+
+  private String rootPath;
+  private final IEnvelopeSender envelopeSender;
+  private ILogger logger;
+
+  // The preferred overload (Taking File instead of String) is only available from API 29
+  @SuppressWarnings("deprecation")
+  EnvelopeFileObserver(String path, IEnvelopeSender envelopeSender, ILogger logger) {
+    super(path);
+    this.rootPath = Objects.requireNonNull(path, "File path is required.");
+    this.envelopeSender = Objects.requireNonNull(envelopeSender, "Envelope sender is required.");
+    this.logger = Objects.requireNonNull(logger, "Logger is required.");
+  }
+
+  @Override
+  public void onEvent(int eventType, @android.support.annotation.Nullable String relativePath) {
```

Comment:
this is probably not required `@android.support.annotation.Nullable` but if so, try to use the one we already have which is `org.jetbrains:annotations`

Reply:
Couldn't choose whether to add it or not, nor the package it comes from. See above, will try again before adding the package to `Config`